src/engine/renderer/tr_image.cpp

```diff
@@ -865,7 +865,7 @@ void R_UploadImage( const byte **dataArray, int numLayers, int numMips,
 	// deal with a half mip resampling
 	if ( image->type == GL_TEXTURE_CUBE_MAP )
 	{
```

Comment:
This is the existing code I was talking about. Instead of creating yet another downscaling mechanism, I suggest using this one by, say, taking the min of picmax and the glconfig's max size value.
Although I think the existing code is broken for compressed textures, since it doesn't set `picmip`.

Reply:
glconfig max size value is not used anymore.